k8s/upgrades/0.8.0-0.8.1/jiva_volume_upgrade.sh

```diff
@@ -61,11 +64,70 @@ sc_res_ver=`kubectl get sc $sc_name -n $ns -o jsonpath="{.metadata.resourceVersi
 # ctrl-dep: pvc-cec8e86d-0bcc-11e8-be1c-000c298ff5fc-ctrl       # 
 #################################################################
 
-c_dep=$(echo $pv-ctrl)
-r_dep=$(echo $pv-rep)
-c_svc=$(echo $c_dep-svc)
-c_name=$(echo $c_dep-con)
-r_name=$(echo $r_dep-con)
+c_dep=$(kubectl get deploy -n $ns -l openebs.io/persistent-volume=$pv,openebs.io/controller=jiva-controller -o jsonpath="{.items[*].metadata.name}")
+r_dep=$(kubectl get deploy -n $ns -l openebs.io/persistent-volume=$pv,openebs.io/replica=jiva-replica -o jsonpath="{.items[*].metadata.name}")
+c_svc=$(kubectl get svc -n $ns -l openebs.io/persistent-volume=$pv -o jsonpath="{.items[*].metadata.name}")
+c_name=$(kubectl get deploy -n $ns $c_dep -o jsonpath="{range .spec.template.spec.containers[*]}{@.name}{'\n'}{end}" | grep "con")
+r_name=$(kubectl get deploy -n $ns $r_dep -o jsonpath="{range .spec.template.spec.containers[*]}{@.name}{'\n'}{end}" | grep "con")
+
+# Fetch the older target and replica - ReplicaSet objects which need to be 
+# deleted before upgrading. If not deleted, the new pods will be stuck in 
+# creating state - due to affinity rules. 
+
+c_rs=$(kubectl get rs -o name --namespace $ns -l openebs.io/persistent-volume=$pv,openebs.io/controller=jiva-controller | cut -d '/' -f 2)
```

Comment:
I was also suggesting this way in my initial comments of previous PR.
Later, found the below command in last time's upgrade script that avoids the 'cut' as well:
```kubectl get rs -n openebs  -l openebs.io/persistent-volume=$pv  -o jsonpath="{range .items[?(@.metadata.ownerReferences[0].name=='$target_dep')]}{@.metadata.name}{end}```

Reply:
you are genius @vishnuitta